packages/next-codemod/bin/upgrade.ts

```diff
@@ -178,7 +178,8 @@ export async function runUpgrade(
   // The following React codemods are for React 19
   if (
     !shouldStayOnReact18 &&
-    compareVersions(targetReactVersion, '19.0.0-beta.0') >= 0
+    compareVersions(targetReactVersion, '18.9999.9999') > 0 &&
```

Comment:
we've already did sth similiar before, this means the segment `major`, `minor` and `patch` are same,  between 2 versions, regardless the suffix

https://github.com/vercel/next.js/blob/3d148d690da02f2f6a713978ddd7e1315fe3cd13/packages/next-codemod/bin/upgrade.ts#L433-L456

Reply:
Can you give me an example where this doesn't work?

Reply:
I didn't say it's not working, we can a not hacky implementation. if you want to go with this hacky number it's also fine

Reply:
18.9999.9999 was not the best choice. Suggesting `19.0.0-0` and making it consistent with other codemods: https://github.com/vercel/next.js/pull/71457